core-models/src/main/scala/com/mesosphere/usi/core/models/PodSpec.scala

```diff
@@ -12,4 +16,42 @@ package com.mesosphere.usi.core.models
   * @param goal target goal of this pod. See [[Goal]] for more details
   * @param runSpec WIP the thing to run, and resource requirements, etc.
   */
-case class PodSpec(id: PodId, goal: Goal, runSpec: RunSpec)
+trait PodSpec {
+  def id: PodId
+  def goal: Goal
+  def runSpec: RunSpec
+}
+
+object PodSpec {
+
+  type ErrorMessage = String
+
+  /**
+    * Verifies that every value in range is requested only once. Requesting same value multiple times would yield pod that will never be scheduled.
+    * E.g. requesting two ports 80 on one machine is just not possible to satisfy
+    * @param runSpec runSpec we are validating
+    * @return true if provided range requirements are valid
+    */
+  private def validateRunSpec(runSpec: RunSpec): Option[ErrorMessage] = {
```

Comment:
I remember talking with @jeschkies about alternatives to wix-validation for Marathon and we both were fond of [Scalactic's](http://www.scalactic.org/user_guide/OrAndEvery) `Good, Bad, One, Many and Every` types. They are pretty simple, yet powerful abstractions especially when it comes to multiple validation errors and their messages. Wdyt?

Reply:
I think the validation is pretty basic for now so we could be fine with a simple option. Also as a library, I am always trying to depend on as small amount of dependencies as possible to make it easier for people integrating.

But I am not against that library if other feel like we absolutely need it to maintain readable validation code :)

Reply:
I'd like to lay down a good pattern for validation. `Option` only brings you so far. The best example is multiple error messages (in case you have multiple errors) - scalactic already elegantly solves this for you.

Reply:
I agree with Alex. `Option`, `Either` and `Try` are not that powerful for validations. There's a reason the Scala ecosystem has a ton. I like Scalactic's for its simplicity.